File: src/main/java/com/chatchatabc/jpademojava/domain/service/JwtService.java
package com.chatchatabc.jpademojava.domain.service;

import com.chatchatabc.jpademojava.domain.model.User;
import org.springframework.stereotype.Service;

@Service
public interface JwtService {

    /**
     * Generate JWT token
     *
     * @param user
     * @return
     */
    String generateToken(User user);

    /**
     * Validate token and get user id
     *
     * @param token
     * @return
     */
    String validateTokenAndGetId(String token) throws Exception;
}

File: src/main/kotlin/com/chatchatabc/jpademo/domain/service/JwtService.kt
package com.chatchatabc.jpademo.domain.service

import com.chatchatabc.jpademo.domain.model.User
import org.springframework.stereotype.Service

@Service
interface JwtService {
    /**
     * Generate JWT Token
     */
    fun generateToken(user: User): String;

    /**
     * Validate JWT Token and get ID
     */
    fun validateTokenAndGetIds(token: String): String;
}